soc/nxp/imxrt/imxrt118x/CMakeLists.txt

```diff
@@ -8,4 +8,10 @@ zephyr_sources(soc.c)
 
 zephyr_include_directories(.)
 
+if(CONFIG_MEMC_MCUX_FLEXSPI)
+  zephyr_sources(./../imxrt11xx/flexspi.c)
+  if(CONFIG_FLASH_MCUX_FLEXSPI_XIP)
+    zephyr_code_relocate(FILES ./../imxrt11xx/flexspi.c LOCATION ${CONFIG_FLASH_MCUX_FLEXSPI_XIP_MEM}_TEXT)
```

Comment:
this dependency is not going to be obvious to someone only looking at rt1170 is my observation about this

Reply:
I have prepared standalone flexspi.c file for RT118x series.